### DISCUSSION TITLE
Protocol Type in State of Reducer

### DISCUSSION THREAD

**Discussion by byunsangun** [pointfreeco/swift-composable-architecture#2890] [Q&A]
~~~Swift
public struct TestProtocol {
    var temp: String { get }
}

struct TestReducer: Reducer {
    struct State: Equatable {
        var temp: TestProtocol
    }
}
~~~

I'm trying to create a field within a State inside a Reducer, 
and I want to declare the type of this field as a protocol. 

However, I'm encountering the following error: 

> "Type 'TestReducer.State' does not conform to protocol 'Equatable'".

 I'm curious about why this error is occurring and how to resolve it.

Thank you in advance.

**Comment by mbrandonw** [comment]
Hi @byunsangun, this is not a TCA problem. It's just how Swift works. Existentials like `any TestProtocol` do not conform to `Equatable` even if `TestProtocol: Equatable`. You just can't use protocols like this if you want your state to be equatable.
